ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFTimestamp.java

```diff
@@ -71,22 +71,20 @@ public class GenericUDFTimestamp extends GenericUDF {
    * otherwise, it's interpreted as timestamp in seconds.
    */
   private boolean intToTimestampInSeconds = false;
-  private boolean strict = true;
 
   @Override
   public ObjectInspector initialize(ObjectInspector[] arguments) throws UDFArgumentException {
     checkArgsSize(arguments, 1, 1);
     checkArgPrimitive(arguments, 0);
     checkArgGroups(arguments, 0, tsInputTypes, STRING_GROUP, DATE_GROUP, NUMERIC_GROUP, VOID_GROUP, BOOLEAN_GROUP);
 
-    strict = SessionState.get() != null ? SessionState.get().getConf()
-        .getBoolVar(ConfVars.HIVE_STRICT_TIMESTAMP_CONVERSION) : new HiveConf()
-        .getBoolVar(ConfVars.HIVE_STRICT_TIMESTAMP_CONVERSION);
     intToTimestampInSeconds = SessionState.get() != null ? SessionState.get().getConf()
         .getBoolVar(ConfVars.HIVE_INT_TIMESTAMP_CONVERSION_IN_SECONDS) : new HiveConf()
         .getBoolVar(ConfVars.HIVE_INT_TIMESTAMP_CONVERSION_IN_SECONDS);
 
-    if (strict) {
+    SessionState ss = SessionState.get();
+
+    if (ss != null && ss.getConf().getBoolVar(ConfVars.HIVE_STRICT_TIMESTAMP_CONVERSION)) {
```

Comment:
The root cause is SessionState is not available at runtime. So, we need to inject the proper config from anywhere else.
For example, we implement `GenericUDF#configure` for some UDFs in case that SessionState is unavailable.
https://github.com/apache/hive/blob/a1420ed816c315d98be7ebf05cdc3ba139a68643/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFToUnixTimeStamp.java#L125-L132
https://github.com/apache/hive/blob/a1420ed816c315d98be7ebf05cdc3ba139a68643/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFRegExp.java#L64-L70